_docs/guides/endpoints.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Install Istio for Google Cloud Endpoints Services
+overview: Explains how to manually integrate Google Cloud Endpoints services with Istio.
+
+order: 42
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This document shows how to manually integrate Istio with existing
+Google Cloud Endpoints services.
+
+## Before you begin
+
+If you don't have an Endpoints service and want to try it out, you can follow
+the instruction to setup an Endpoints service on GKE
+[instructions](https://cloud.google.com/endpoints/docs/openapi/get-started-kubernetes-engine).
+After setup, you should be able to get an API key and store it in `ENDPOINTS_KEY` environment variable and the external IP address `EXTERNAL_IP`.
+You may test the service using the following command:
+```bash
+curl --request POST --header "content-type:application/json" --data '{"message":"hello world"}' "http://${EXTERNAL_IP}:80/echo?key=${ENDPOINTS_KEY}"
+```
+
+You need to install Istio with [instructions]({{home}}/docs/setup/kubernetes/quick-start.html#google-kubernetes-engine).
+
+## HTTP Endpoints service
+
+1. Inject the service into mesh using `--includeIPRanges` so that Egress is allowed to call external services directly.
+Otherwise, ESP won't be able to access Google cloud service control. Follow the [instructions]({{home}}/docs/tasks/traffic-management/egress.html#calling-external-services-directly).
+
+1. After injection, issue the same test command as above to insure calling ESP continues to work.
+
+1. If you want to access the service through Ingress, create the following ingress service:
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: simple-ingress
+  annotations:
+    kubernetes.io/ingress.class: istio
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /echo
+        backend:
+          serviceName: esp-echo
+          servicePort: 80
+EOF
+```
+
+1. Get the Ingress IP through [instructions]({{home}}/docs/tasks/traffic-management/ingress.html#verifying-http-ingress).
+You can verify accessing the Endpoints service through Ingress:
+```bash
+curl --request POST --header "content-type:application/json" --data '{"message":"hello world"}' "http://${INGRESS_HOST}:80/echo?key=${ENDPOINTS_KEY}"i
+```
+
+## HTTPS Endpoints service using secured Ingress
+
+The recommended way to securely access a mesh Endpoints service is through secured Ingress with a mTLS enabled mesh.
+
+1. Expose HTTP port in your mesh service.
+Adding `"--http_port=8081"` in the ESP deployment arguments and expose a HTTP port through Service ports:
+```yaml
+  - port: 80
+    targetPort: 8081
+    protocol: TCP
+    name: http
+```
+Update the mesh service deployment.
+
+1. Turn on mTLS in Istio. By using the following command:
+```bash
+kubectl edit cm istio -n istio-system
+```
+
+And uncomment the line:
```

Comment:
This line and the following code block need to be indented by 3 spaces so that they appear under the above bullet and then the numbered list will show up properly. If you look at the rendered page, you'll see that the numbered list is not rendered right.

Reply:
Fixed. PTAL.